c++/src/kj/async-io-test.c++

```diff
@@ -3574,5 +3576,45 @@ KJ_TEST("pump file to socket") {
   doTest(fs->getCurrent().createTemporary());
 }
 
+KJ_TEST("Calling abortRead() while tryRead() is in progress") {
+  // This is a stream that will only permit one call to tryRead() to be in progress at any one time
+  // Any tryRead calls won't ever complete
+  class NonConcurrentStream: public kj::AsyncInputStream {
+    public:
+    virtual kj::Promise<size_t> tryRead(void* buffer, size_t minBytes, size_t maxBytes) override {
+      numConcurrentCalls.fetch_add(1);
+      co_await (kj::Promise<void>) kj::NEVER_DONE;
+      numConcurrentCalls.fetch_sub(1);
```

Comment:
This line will never execute since the previous line will never return normally -- exceptions and cancellation will skip this. But you probably want to decrement the counter in those cases, right? You can move this up and wrap it in KJ_DEFER() to make sure it runs no matter how the scope is exited.